src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java

```diff
@@ -33,21 +44,49 @@
     private final HearingRecordingSegmentRepository segmentRepository;
     private final BlobstoreClient blobstoreClient;
     private final AuditEntryService auditEntryService;
+    private final ShareesRepository shareesRepository;
+    private final SecurityService securityService;
+
+
+    @Value("${shareelink.ttl}")
+    int validityInHours;
 
     @Autowired
     public SegmentDownloadServiceImpl(HearingRecordingSegmentRepository segmentRepository,
-                                      BlobstoreClient blobstoreClient, AuditEntryService auditEntryService) {
+                                      BlobstoreClient blobstoreClient, AuditEntryService auditEntryService,
+                                      ShareesRepository shareesRepository,
+                                      SecurityService securityService) {
         this.segmentRepository = segmentRepository;
         this.blobstoreClient = blobstoreClient;
         this.auditEntryService = auditEntryService;
+        this.shareesRepository = shareesRepository;
+        this.securityService = securityService;
     }
 
 
     @Override
-    public HearingRecordingSegment fetchSegmentByRecordingIdAndSegmentNumber(UUID recordingId, Integer segmentNo) {
+    public HearingRecordingSegment fetchSegmentByRecordingIdAndSegmentNumber(UUID recordingId, Integer segmentNo,
+                                                                             String userToken) {
 
         HearingRecordingSegment segment =
             segmentRepository.findByHearingRecordingIdAndRecordingSegment(recordingId, segmentNo);
+
+
+        //Check if user access has expired
+        String userEmail = securityService.getUserEmail(userToken);
+        List<HearingRecordingSharee> hearingRecordingSharees = shareesRepository.findByShareeEmail(userEmail);
+        if (CollectionUtils.isNotEmpty(hearingRecordingSharees)) {
+            Optional<HearingRecordingSharee> recordingSharee = hearingRecordingSharees.stream()
+                .filter(hearingRecordingSharee ->
+                            getHearingRecordingShareeSegment(hearingRecordingSharee.getHearingRecording(), segmentNo))
```

Comment:
is segment no unique throughout the DB ? I could not see we do not match recording ID to user.